### DISCUSSION TITLE
[3.2] FR: Add "entries per page" setting to Entry indexes

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#8707] [Ideas]
### Description

It would be nice if it was possible to override the default limit of 50 entries per page to a different number, on a per-source basis in the Entry index (i.e. a dropdown or number input beside the pagination UI).

### Steps to reproduce

1.
2.

### Additional info

- Craft version: 3.2
- PHP version:
- Database driver & version:
- Plugins & versions:

**Comment by ul8** [comment]
I'd be happy with just a global setting, if that helps to get this done earlier. The per-source basis feature can still be added later...but the fixed limit of 50 is way too small for my usual needs.

**Comment by ryanpcmcquen** [comment]
@brandonkelly, is it configurable at all? 100 seems like a better default than 50 ...

**Comment by brandonkelly** [comment]
Agree with that. Just increased the page/batch size to 100 for the next 3.4 release.

**Comment by owldesign** [comment]
You can also try `Craft.elementIndex.settings.batchSize = 1000` to update batchSize limit. Run that javascript on the page where your element index is.

**Comment by OwenMelbz** [comment]
Is there a definitive way to configure this?

We're getting bad performance issues when loading 100 of them in, we were actually happy with only showing 50 max as if the user needed to find something, sifting through 100 items is a chore when you can just use the search to find it.

**Reply by brandonkelly** [reply]
Use the [Control Panel JS](https://plugins.craftcms.com/cp-js) plugin to register this JS code:

```js
Craft.elementIndex.settings.batchSize = 50;
```

**Reply by gixxy22** [reply]
Ive tried this:

$(function(){
    Craft.elementIndex.settings.batchSize = 5;
});

however the first page load is the default of 100, any next pages are 5. how can i set the initial page size?

**Reply by brandonkelly** [reply]
@mikemeadeuk You probably need to register it via an external JS file, so it loads before the element index is instantiated.

**Reply by weotch** [reply]
When I add `Craft.elementIndex.settings.batchSize = 50;` to an external JS file registered with Control Panel JS, I get errors because `Craft.elementIndex` is undefined.  

![image](https://user-images.githubusercontent.com/77567/193346087-69ea5f2a-28da-4159-8c61-994760e3f8a9.png)

If I wrap it in `$()`, like...

```js
$(function() {
	Craft.elementIndex.settings.batchSize = 50;
})
```

... then it fires too late and doesn't affect the initial page of results.  I would love another recommendation on how to set the initial batchSize.

My use case is the same as @OwenMelbz, the performance of the entry listing page is poor with a number of sites and entries (like 35s to return the `admin/actions/element-indexes/get-elements` XHR response in Craft 3.7.55.2).  Looking at the debug bar, it looks like the amount of queries run is based on how many results are returned.  When I force the `batchSize` count to 10, that 35s changes to 6s.

**Reply by JeanLucEsser** [reply]
FWIW, I need this as well but for the opposite reason, need to show all 300 elements to be able to export them in a single shot. A configurable global setting would go a long way, a per section option would be glorious. Craft 4 here.

**Reply by brianjhanson** [reply]
Give `Craft.BaseElementIndex.defaults.batchSize = 50` a try. That should work without having to wrap in a document.ready kind of thing. Be warned, it will set the default for all element indexes so it might be a little heavy handed.

**Comment by arifje** [comment]
Maybe I am not seeing it, but why is this still not a simple config variable which we can set in the general config? 

The JS injection with the plugin works, but only when the elemets are loaded async (clicking a filter or searching for entries), not on first page load. Which still results in an index with 100 elements in my case, but with wrong pagination.

Screenshot; 

![Screenshot 2025-03-25 at 14 25 18](https://github.com/user-attachments/assets/21a7c8c9-f0d8-4e71-bf37-334f42bbc490)

Running Craft 4.14.10

**Comment by MatthiasBrodelet** [comment]
I am running into the same issue.
We would like to increase the amount of entries per page.

Why is it not possible to choose a number of rows / entries per page just like it is possible in a MUI data table?
Example: https://mui.com/material-ui/react-table/#data-table

For a CMS like Craft this seems like a really weird limitation.
